mono/tests/test-runner.cs

```diff
@@ -219,6 +238,14 @@ class TestInfo {
 							Console.Write (".");
 						passed.Add(process_data [dead]);
 						npassed ++;
+					// FIXME: is this valid for all targets?
+					} else if (dead.ExitCode == 137) {
```

Comment:
C# doesn't control that, it just returns whatever the OS gives it. POSIX specifies that the exit code for signals is 128+n, for SIGKILL n=9 which results in 137.